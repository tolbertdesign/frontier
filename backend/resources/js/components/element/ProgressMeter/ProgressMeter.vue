<template>
  <div :class="['meter', backgroundColorClass, (showValue && firstMeterPercentage === 0 ? 'noValueDisplay pb-6': '')]">
    <span
      :style="{ width: firstMeterPercentage + '%' }"
      :class="['meter-1', color === 'secondary' ? 'has-background-secondary': 'bg-primary' ]"
    />
    <span
      v-if="showValue"
      :style="getPercentagePosition(firstMeterPercentage)"
      class="valueDisplay font-semibold"
    >
      {{ valueToDisplay }}
    </span>
    <span
      :style="{ width: secondMeterPercentage + '%' }"
      class="meter-2"
    />
  </div>
</template>

<script>
export default {
  name: 'ProgressMeter',
  status: 'prototype',
  release: '1.0.0',
  props: {
    color: {
      type: String,
      default: 'primary',
    },
    backgroundColorClass: {
      type: String,
      default: 'bg-grey-light',
    },
    goal: {
      type: [String, Number],
      default: 100,
    },
    firstMeter: {
      type: Number,
      default: 0,
    },
    secondMeter: {
      type: Number,
      default: 0,
    },
    showValue: {
      type: Boolean,
      default: false,
    },
    valueToDisplay: {
      type: String,
      default: '',
    },
  },
  computed: {
    firstMeterPercentage () {
      return this.getPercentage(this.firstMeter, this.goal)
    },
    secondMeterPercentage () {
      return this.getPercentage(this.firstMeter + this.secondMeter, this.goal)
    },
  },
  methods: {
    getPercentage (part, total) {
      if (isNaN(part) || isNaN(total) || total <= 0) {
        return 0
      }
      const percent = part / total * 100
      if (percent > 100) {
        return 100
      }
      return percent
    },
    getPercentagePosition (percent) {
      if (percent > 15) {
        return { right: Math.abs((percent - 100)) + '%' }
      } else {
        return { left: '0%' }
      }
    },
  },
}
</script>
