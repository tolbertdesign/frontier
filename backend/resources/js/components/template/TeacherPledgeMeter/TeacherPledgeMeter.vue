<template>
  <div class="px-4 pt-6">
    <h2 class="mb-2 text-2xl font-semibold capitalize">
      {{ classroomName }} - {{ gradeName }} {{ lang.team_pom }}
    </h2>
    <p class="mb-4">
      {{ lang.team_progress }}
    </p>

    <div class="mb-4">
      <ProgressMeter
        :first-meter="pledgedTotal"
        :goal="pledgeGoal"
      />
      <div class="mt-2">
        <h3 class="font-bold">
          {{ lang.total_pledged }}: ${{ pledgedTotal | currency(true) }} of ${{ pledgeGoal | currency(true) }}
        </h3>
        <h4
          v-if="program.program_pledge_setting.flat_donate_only != true"
          class="capitalize"
        >
          {{ program.unit.modifier }} {{ program.unit.name }}
        </h4>
      </div>

      <div class="flex items-center justify-center mt-4">
        <button
          class="w-64 px-12 py-3 text-white rounded-full bg-secondary focus:outline-none focus:shadow-outline"
          @click="getReport"
        >
          {{ lang.pledge_report }}
        </button>
      </div>

    </div>
  </div>
</template>

<script>

export default {
  name: 'TeacherPledgeMeter',
  status: 'prototype',
  release: '1.0',
  props: {
    program: {
      type: Object,
      default: null,
    },
  },
  computed: {
    lang () {
      return this.$store.state.lang
    },
    classroomName () {
      return this.$store.state.User.class_last_name
    },
    participant () {
      const teacherParticipantId = this.$store.state.User.teacher_participant_id
      return this.program.participants.find(participant => participant.id === teacherParticipantId)
    },
    gradeName () {
      return this.participant.participant_info.classroom.grade.display_name
    },
    pledgedTotal () {
      return this.$store.state.User.class_pledge_total
    },
    pledgeGoal () {
      return this.participant.participant_info.classroom.pledge_meter
    },
  },
  methods: {
    getReport () {
      this.$axon.get('report/class-pledge', () => {})
        .then((response) => {
          window.open(response.data.url, '_blank')
        })
    },
  },
}
</script>
