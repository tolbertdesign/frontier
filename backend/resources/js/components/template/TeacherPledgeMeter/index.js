import TeacherPledgeMeter from './TeacherPledgeMeter'

export default TeacherPledgeMeter
