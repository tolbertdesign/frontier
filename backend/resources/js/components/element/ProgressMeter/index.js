import ProgressMeter from './ProgressMeter.vue'

export default ProgressMeter
