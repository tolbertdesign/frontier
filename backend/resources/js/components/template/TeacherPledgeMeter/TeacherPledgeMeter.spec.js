import { shallowMount, createLocalVue } from '@vue/test-utils'
import TeacherPledgeMeter from './TeacherPledgeMeter.vue'
import ProgressMeter from '@/components/element/ProgressMeter'

const localVue = createLocalVue()
localVue.filter('currency', jest.fn())
localVue.component('ProgressMeter', ProgressMeter)

const $store = {
  state: {
    User: {
      class_last_name: 'Test',
    },
    lang: {
      student_star_video: {},
    },
  },
  watch: jest.fn(),
}

const program = {
  unit: {
    modifier: 'per',
    name: 'Reading Challenge',
  },
  program_pledge_setting: {
    flat_donate_only: 0,
  },
  participants: [
    {
      profile: {
        pledge_goal: 100,
      },
      participant_info: {
        classroom: {
          grade: {
            display_name: 'Classroom Grade Display Name',
          },
        },
      },
    },
  ],
}

describe('TeacherPledgeMeter', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(TeacherPledgeMeter, {
      localVue,
      mocks: { $store },
      propsData: { program },
    })
    expect(wrapper).toMatchSnapshot()
  })
})
