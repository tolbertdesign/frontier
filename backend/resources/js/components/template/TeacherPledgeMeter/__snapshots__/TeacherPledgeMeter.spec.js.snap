// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeacherPledgeMeter matches snapshot 1`] = `
<div class="px-4 pt-6">
  <h2 class="mb-2 text-2xl font-semibold capitalize">
    Test - Classroom Grade Display Name
  </h2>
  <p class="mb-4">

  </p>
  <div class="mb-4">
    <progressmeter-stub color="primary" backgroundcolorclass="bg-grey-light" goal="100" firstmeter="0" secondmeter="0" valuetodisplay=""></progressmeter-stub>
    <div class="mt-2">
      <h3 class="font-bold">
        : $ of $
      </h3>
      <h4 class="capitalize">
        per Reading Challenge
      </h4>
    </div>
    <div class="flex items-center justify-center mt-4"><button class="w-64 px-12 py-3 text-white rounded-full bg-secondary focus:outline-none focus:shadow-outline">

      </button></div>
  </div>
</div>
`;
